libs/state/spec/core/utils/safe-pluck.spec.ts

```diff
@@ -1,22 +1,34 @@
 import { safePluck } from '../../../src/lib/core/utils';
 
+interface ISafePluckSpec {
+  foo: { bar: string };
+}
 describe('safePluck', () => {
-    const bar = 'hello'
-    const obj = { foo: { bar } };
-    it('should not touch passed arguments', () => {
-        const args = ['foo', 'bar'];
-        safePluck(obj, args);
-        expect(args.length).toBe(2);
-    })
-    it('should return `bar` when keys are provided', () => {
-        expect(safePluck(obj, ['foo', 'bar'])).toEqual(bar);
-    })
+  const bar = 'hello';
+  const obj: ISafePluckSpec = { foo: { bar } };
+  it('should not touch passed arguments', () => {
+    const args = ['foo', 'bar'] as any;
+    safePluck(obj, args);
+    expect(args.length).toBe(2);
+  });
 
-    it('should return whole `obj` when keys are not provided', () => {
-        expect(safePluck(obj, [])).toEqual(obj);
-    })
+  it('should return `bar` when keys are provided', () => {
+    expect(safePluck(obj, ['foo', 'bar'])).toEqual(bar);
+  });
 
-    it('should return whole `obj` when incorrect keys are provided', () => {
-        expect(safePluck(obj, ['incorrect-key'])).toEqual(obj);
-    })
-})
+  it('should return undefined when object is not valid', () => {
+    expect(safePluck([undefined] as any, ['prop1'] as any)).toEqual(undefined);
+  });
+
+  it('should return undefined when keys are not provided', () => {
+    expect(safePluck(obj, null as any)).toEqual(undefined);
+  });
+
+  it('should return undefined when incorrect keys are provided', () => {
+    expect(safePluck(obj, ['incorrect-key'] as any)).toEqual(undefined);
+  });
+
+  it('should return undefined when incorrect keys are provided', () => {
+    expect(safePluck(obj, ['incorrect-key'] as any)).toEqual(undefined);
+  });
```

Comment:
```suggestion
  describe('edge cases', () => {
    it('should return undefined when object is not valid', () => {
      expect(safePluck([undefined] as any, ['prop1'] as any)).toEqual(undefined);
      expect(safePluck(undefined as any, ['prop1'] as any)).toEqual(undefined);
      expect(safePluck('' as any, ['prop1'] as any)).toEqual(undefined);
      expect(safePluck(0 as any, ['prop1'] as any)).toEqual(undefined);
      expect(safePluck([], ['length'])).toEqual(undefined);
      expect(safePluck({} as any, ['prop1'] as any)).toEqual(undefined);
    });

    it('should return null when object is null', () => {
      expect(safePluck(null as any, ['prop1'] as any)).toEqual(null);
      expect(safePluck(null as any, [] as any)).toEqual(null);
      expect(safePluck(null as any, null as any)).toEqual(null);
    });

    it('should return undefined when keys are invalid', () => {
      expect(safePluck(obj, null as any)).toEqual(undefined);
      expect(safePluck(obj, [{}] as any)).toEqual(undefined);
      expect(safePluck(obj, [0] as any)).toEqual(undefined);
      expect(safePluck(obj, [''] as any)).toEqual(undefined);
      expect(safePluck(obj, [undefined] as any)).toEqual(undefined);
      expect(safePluck(obj, [] as any)).toEqual(undefined);
    });
  });
```

Reply:
adapted test cases to really state what is getting returned and making sure every edge case is tested